packages/wallets/src/components/FlowProvider/FlowProvider.tsx

```diff
@@ -0,0 +1,122 @@
+import React, {
+    createContext,
+    ReactElement,
+    ReactFragment,
+    ReactNode,
+    ReactPortal,
+    useContext,
+    useMemo,
+    useState,
+} from 'react';
+import { Formik, FormikErrors, FormikValues } from 'formik';
+
+export type TFlowProviderContext<T> = {
+    WalletScreen?: ReactNode;
+    currentScreenId: keyof T;
+    formValues: FormikValues;
+    isFinalScreen: boolean;
+    setFormValues: (
+        field: string,
+        value: unknown,
+        shouldValidate?: boolean | undefined
+    ) => Promise<FormikErrors<unknown> | void>;
+    switchScreen: (screenId: keyof T) => void;
+    switchNextScreen: () => void;
+};
+
+type FlowChildren = ReactElement | ReactFragment | ReactPortal;
+
+export type TWalletScreens = {
+    [id: string]: ReactNode;
+};
+
+export type TFlowProviderProps<T> = {
+    children: (context: TFlowProviderContext<T>) => FlowChildren;
+    initialValues: FormikValues;
+    screens: T;
+    screensOrder: (keyof T)[];
+};
+
+const FlowProviderContext = createContext<TFlowProviderContext<TWalletScreens> | null>(null);
+
+/**
```

Comment:
I really like that type of docs - that should be a standard. Especially for non-trivial components like this. Without it, I would spend much more time analysing and understanding the flow. 
Minor - might be worth adding a bit explanation what is responsiblity of this component.